pkg/combustion/kubernetes_integration_test.go

```diff
@@ -0,0 +1,122 @@
+//go:build integration
+
+package combustion
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"gopkg.in/yaml.v3"
+)
+
+func TestConfigureManifestsValidDownload(t *testing.T) {
+	// Setup
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	ctx.ImageDefinition.Kubernetes.Manifests.URLs = []string{
+		"https://k8s.io/examples/application/nginx-app.yaml",
+	}
+
+	downloadedManifestsDestDir := filepath.Join(ctx.CombustionDir, manifestsDir)
+	expectedDownloadedFilePath := filepath.Join(downloadedManifestsDestDir, "dl-manifest-1.yaml")
+
+	// Test
+	err := configureManifests(ctx)
+
+	// Verify
+	require.NoError(t, err)
+	assert.FileExists(t, expectedDownloadedFilePath)
+
+	b, err := os.ReadFile(expectedDownloadedFilePath)
+	require.NoError(t, err)
+
+	contents := string(b)
+	assert.Contains(t, contents, "apiVersion: v1")
+	assert.Contains(t, contents, "name: my-nginx-svc")
+	assert.Contains(t, contents, "type: LoadBalancer")
+	assert.Contains(t, contents, "image: nginx:1.14.2")
+}
+
+func TestConfigureKubernetes_SuccessfulRKE2ServerWithManifestURLs(t *testing.T) {
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	ctx.ImageDefinition.Kubernetes = image.Kubernetes{
+		Version: "v1.29.0+rke2r1",
+	}
+	ctx.KubernetesScriptInstaller = mockKubernetesScriptInstaller{
+		installScript: func(distribution, sourcePath, destPath string) error {
+			return nil
+		},
+	}
+	ctx.KubernetesArtefactDownloader = mockKubernetesArtefactDownloader{
+		downloadArtefacts: func(arch image.Arch, version, cni string, multusEnabled bool, destPath string) (string, string, error) {
+			return serverInstaller, serverImages, nil
+		},
+	}
+
+	ctx.ImageDefinition.Kubernetes.Manifests.URLs = []string{
+		"https://k8s.io/examples/application/nginx-app.yaml",
```

Comment:
It doesn't seem right with me at all that we have to perform an actual download here instead of mocking the call. I guess this can land as is but we desperately need an interface behind the registry calls.